mautic/mautic
# Set the page language for the preference center page if the lead's preferred locale is not set

…ead preferred locale



| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:


This PR offer to use the page language as a language of the preference center page.

The following process is used to set language for the preference center page: `preferred contact locale --> user locale -> system locale`
This PR offers the next one: `preferred contact locale --> page language -> user locale -> system locale`
#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a contact and don't set the `Preferred Locale` field
3. Create a Landing Page as a preference center. Add the {preferredchannel}, {saveprefsbutton} variables to the Landing Page.
4. Create a segment email with the preference center configured and use the unsubscribe link in it. 
5. Choose the created landing page as the Preference Center Page.
6. Send the email
7. Open the email
8. Click the unsubscribe link
9. Check if the selected page language is used
10. Set the  `Preferred Locale` field for the contact
11. Reload the preference center page
12. Check if the prefered contact language is used
